src/ops.rs

```diff
@@ -442,6 +456,9 @@ impl KeyEncryptable for Message {
         let ke_ka = kmac_xof(&Z.x.to_bytes().to_vec(), &[], 448 * 2, "PK", d)?;
         let (ke, ka) = ke_ka.split_at(ke_ka.len() / 2);
 
+        // Create a copy of the original message
+        let original_msg = self.msg.clone(); // can nefarious actors access this clone?
```

Comment:
you do not want to do this, cloning should be avoided wherever possible. the goal is to be able to handle very large amounts of data and this creates a copy on the heap. you appear to be correctly handling this below. it doesnt appear that youre even using this copy so it should be removed.